rmw_fastrtps_shared_cpp/src/rmw_node.cpp

```diff
@@ -363,6 +364,33 @@ __rmw_destroy_node(
   return result_ret;
 }
 
+rmw_ret_t
+__rmw_node_assert_liveliness(
+  const char * identifier,
+  const rmw_node_t * node)
+{
+  RMW_CHECK_ARGUMENT_FOR_NULL(node, RMW_RET_INVALID_ARGUMENT);
+  RMW_CHECK_TYPE_IDENTIFIERS_MATCH(
+    node,
+    node->implementation_identifier,
+    identifier,
+    return RMW_RET_INCORRECT_RMW_IMPLEMENTATION);
+
+  auto node_info = static_cast<CustomParticipantInfo *>(node->data);
```

Comment:
This will cast away the const from rmw_node_t (this happens everywhere but is technically incorrect). We may want to be using rmw_node_t * const node to show our actual usage.

Reply:
How about `static_cast<const CustomParticipantInfo *>` instead?